modApiServer/src/org/aion/api/server/rpc/RpcProcessor.java

```diff
@@ -9,16 +9,22 @@
 import org.slf4j.Logger;
 
 import java.util.List;
-import java.util.concurrent.TimeUnit;
+import java.util.concurrent.*;
 
 public class RpcProcessor {
 
     private static final Logger LOG = AionLoggerFactory.getLogger(LogEnum.API.name());
 
-    RpcMethods apiHolder;
+    private RpcMethods apiHolder;
+
+    private ExecutorService executor;
+    private CompletionService<JSONObject> batchCallCompletionService;
+    private final int SHUTDOWN_WAIT_SECONDS = 5;
 
     public RpcProcessor(List<String> enabled) {
         this.apiHolder = new RpcMethods(enabled);
+        executor = Executors.newFixedThreadPool(Math.min(Runtime.getRuntime().availableProcessors() * 2, 4));
```

Comment:
@ali-sharif 
feel like we open ThreadPool in the different place within rpc server and the rpc processor
Can we manage these thread in the same place?

Reply:
Those two thread pools serve a different purpose. 

The ***existing*** thread pool in the RPC server is HTTP server implementation-specific (ie. undertow manages it's own thread pool, we have to manage Nano's thread pool, etc.). 

This ***new*** thread pool deals with parallelizing the processing of the RPC batch calls and I think this is the most appropriate place to allocate this thread pool, since this class is delegated the responsibility of handling the RPC call String payload by the HTTP server.

Let me know if there is another place in the class hierarchy where you think might be a more appropriate place for this thread pool to be instantiated.